### PR TITLE
DAOS-14666 golang update to 1.21.4

### PR DESCRIPTION
**Pull request by ryon-jensen**
Signed-off-by: Ryon Jensen <ryon.jensen@intel.com>

### CODE REVIEW

File: go.spec
Comment (brianjmurrell) [inline go.spec:1]: You should also add a `%changelog` entry and reset the `Release:` back to `1` to reflect the `Version:` update.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Comment by brianjmurrell** [comment]
Looks good.  But your Leap 15.5 build failure is probably because you are working in the midst of an O/S update.

Try checking out https://github.com/daos-stack/packaging/pull/77 into a `$directory` somewhere and then from the top dir of the workspace for this PR run `$directory/install`.  That should update the `packaging/` directory contents for this PR's workspace.  Commit that and push it and see if that resolves the Leap 15 build.

**Comment by ryon-jensen** [comment]
> Looks good. But your Leap 15.5 build failure is probably because you are working in the midst of an O/S update.
> 
> Try checking out [daos-stack/packaging#77](https://github.com/daos-stack/packaging/pull/77) into a `$directory` somewhere and then from the top dir of the workspace for this PR run `$directory/install`. That should update the `packaging/` directory contents for this PR's workspace. Commit that and push it and see if that resolves the Leap 15 build.

Did as instructed. Got the following errors, but I imagine they are not an issue:
```ln: failed to create hard link '.git/hooks/applypatch-msg.sample': File exists
ln: failed to create hard link '.git/hooks/commit-msg.sample': File exists
ln: failed to create hard link '.git/hooks/fsmonitor-watchman.sample': File exists
ln: failed to create hard link '.git/hooks/post-update.sample': File exists
ln: failed to create hard link '.git/hooks/pre-applypatch.sample': File exists
ln: failed to create hard link '.git/hooks/pre-commit.sample': File exists
ln: failed to create hard link '.git/hooks/pre-merge-commit.sample': File exists
ln: failed to create hard link '.git/hooks/prepare-commit-msg.sample': File exists
ln: failed to create hard link '.git/hooks/pre-push.sample': File exists
ln: failed to create hard link '.git/hooks/pre-rebase.sample': File exists
ln: failed to create hard link '.git/hooks/pre-receive.sample': File exists
ln: failed to create hard link '.git/hooks/push-to-checkout.sample': File exists
ln: failed to create hard link '.git/hooks/update.sample': File exists
```

**Comment by brianjmurrell** [comment]
Yes, those can be ignored.

**Comment by brianjmurrell** [comment]
This PR is going to be blocked by #13117 and #13390, (a) so that the O/S is upgraded after adding the repos causing the go packages to actually get upgraded to the ones from this PR and (b) that PR repos and our Artifactory repo will be candidates for go package upgrades.

Those both need to land to `master` and `release/2.4` in order for testing in this PR to be complete and accurate.

#13390 is waiting for reviews and as you know, #13117 is blocked on the raft issues.

**Comment by ryon-jensen** [comment]
Test failures in the daos master build aren't related.

**Review by JohnMalmberg — DISMISSED** [review]

**Review by JohnMalmberg — APPROVED** [review]

**Pull request merged**